src/textOnlyPackageGenerator/PackageGenerator.proj

```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+        
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <TargetPackagesPath>$(ArtifactsDir)textOnlyPackages/</TargetPackagesPath>
+    <CsProjTemplatePath>./textOnlyPackage.csproj.template</CsProjTemplatePath>
+  </PropertyGroup>
+
+  <Import Project="../common/RestoreProjects.proj"/>
+
+  <Target Name="GetPackageInfo" AfterTargets="Restore">
+    <ReadLinesFromFile File="$(PathToCsv)">
+      <Output TaskParameter="Lines" ItemName="CsvLines"/>
+    </ReadLinesFromFile>
+    
+    <ItemGroup>
+      <PackageInfo Include="@(CsvLines)" Exclude="">
+        <PackageName>$([System.String]::Copy('%(CsvLines.Identity)').Split(',')[0])</PackageName>
+        <PackageVersion>$([System.String]::Copy('%(CsvLines.Identity)').Split(',')[1])</PackageVersion>
+      </PackageInfo>
+    </ItemGroup>
+  </Target>
+    
+  <!-- see ./generate-text-packs.sh for `AfterTargets` justification -->
+  <Target Name="CopyTextOnlyPackages" AfterTargets="Restore">
+    <MSBuild
+      Projects="$(MSBuildProjectFile)"
+      Targets="CopyTextOnlyPackage"
+      Properties="PackageName=%(PackageInfo.PackageName);PackageVersion=%(PackageInfo.PackageVersion)"/>
+  </Target>
+    
+  <Target Name="CopyTextOnlyPackage">
+    <PropertyGroup>
+      <RestoredPackagePath>$(TargetPackagesPath)$(PackageName.ToLower())</RestoredPackagePath>
+      <TextOnlyPackageDirectory>$(PackageName.ToLower())</TextOnlyPackageDirectory>
+      <CsProjFileName>$(PackageName.ToLower()).$(PackageVersion).csproj</CsProjFileName>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <TextOnlyPackageContent
+        Include="$(RestoredPackagePath)/**/*"
+        Exclude="
+          $(RestoredPackagePath)/**/.nupkg.metadata;
+          $(RestoredPackagePath)/**/.signature.p7s;
+          $(RestoredPackagePath)/**/*.nupkg;
+          $(RestoredPackagePath)/**/*.nupkg.sha512;"/>
```

Comment:
Why do we exclude some here and some below?

Reply:
Here we explicitly remove metadata assets we know we don't need. Below we verify that there are no assets in the package that we don't know of. If any are found - we raise an exception. We don't remove any file there